core/arch/arm/kernel/generic_entry_a32.S

```diff
@@ -151,9 +151,21 @@ END_FUNC wait_secondary
 LOCAL_FUNC reset , :
 UNWIND(	.fnstart)
 UNWIND(	.cantunwind)
+
+/*
+ * if built-in arguments are configured, no arguments are expected from
+ * loader; instead, required parameters are configured in build time.
+ */
+
+#if defined (CFG_TEE_BUILT_IN_ARGS)
+    ldr r4, =CFG_TEE_PAGEABLE_ADDR
```

Comment:
The harm with "built-in argument" together with ARM-TF is the it basically puts a requirement on ARM-TF to __never__ change the arguments for that particular platform. That's a requirement which isn't needed as the arguments always are supplied any way. Please add something like:
```
ifeq($(CFG_TEE_BUILT_IN_ARGS),y)
ifeq($(CFG_WITH_ARM_TRUSTED_FW),y)
$(error error: CFG_TEE_BUILT_IN_ARGS is incompatible with CFG_WITH_ARM_TRUSTED_FW)
endif
endif
```
to `core/arch/arm/arm.mk`

By the way, this code is used on ARMv8 too, when running in Aarch32.